File: src/algorithms/max-power.js
/**
 * Given a string s, returns the power of s.
 * 
 * The power of the string is the maximum length of a non-empty substring that
 * contains only one unique character.
 * 
 * @param {string} s - A string of lowercase English characters
 * @returns {number} - Returns a count of the longest substring of one character
 */
function maxPower (s) {
  let res = 0;
  let temp = 0;
  let current = s[0];
  for (let i = 0; i < s.length; i++) {
    if (s[i] === current) {
      temp++;
    } 
    if (temp > res) {
      res = temp;
    }
    if (s[i] !== current) {
      current = s[i];
      temp = 1;
    }
  }
  return res;
}

File: src/algorithms/max-power.py
def max_power(s):
    """A function to determine the power of a string.

    The power of the string is the maximum length of a non-empty substring that contains only one unique character. Given a string s, return the power of s.

    Args:
        s (String): A string consisting of only lowercase English letters.

    Returns:
        Integer: Returns an integer denoting the power of s.
    """
    max_count = 1
    temp_count = 1
    length = len(s)
    if length == 1:
        return length
    for i in range(1, length):
        if s[i] == s[i - 1]:
            temp_count += 1
            max_count = max(temp_count, max_count)
        else:
            max_count = max(temp_count, max_count)
            temp_count = 1
    return max_count
